bot/exts/moderation/silence.py

```diff
@@ -93,87 +93,222 @@ async def _async_init(self) -> None:
         await self.bot.wait_until_guild_available()
 
         guild = self.bot.get_guild(Guild.id)
-        self._verified_role = guild.get_role(Roles.verified)
+
+        self._verified_msg_role = guild.get_role(Roles.verified)
+        self._verified_voice_role = guild.get_role(Roles.voice_verified)
+        self._helper_role = guild.get_role(Roles.helpers)
+
         self._mod_alerts_channel = self.bot.get_channel(Channels.mod_alerts)
+
         self.notifier = SilenceNotifier(self.bot.get_channel(Channels.mod_log))
         await self._reschedule()
 
+    async def _get_related_text_channel(self, channel: VoiceChannel) -> Optional[TextChannel]:
+        """Returns the text channel related to a voice channel."""
+        # TODO: Figure out a dynamic way of doing this
+        channels = {
+            "off-topic": Channels.voice_chat,
+            "code/help 1": Channels.code_help_voice,
+            "code/help 2": Channels.code_help_voice,
+            "admin": Channels.admins_voice,
+            "staff": Channels.staff_voice
+        }
+        for name in channels.keys():
+            if name in channel.name.lower():
+                return self.bot.get_channel(channels[name])
+
+    async def send_message(self, message: str, source_channel: TextChannel,
+                           target_channel: Union[TextChannel, VoiceChannel],
+                           alert_target: bool = False, duration: HushDurationConverter = 0) -> None:
+        """Helper function to send message confirmation to `source_channel`, and notification to `target_channel`."""
+        await source_channel.send(
+            message.replace("current", target_channel.mention if source_channel != target_channel else "current")
+                   .replace("{duration}", str(duration))
+        )
+
+        voice_chat = None
+        if isinstance(target_channel, VoiceChannel):
+            # Send to relevant channel
+            voice_chat = await self._get_related_text_channel(target_channel)
+
+        if alert_target and source_channel != target_channel:
+            if isinstance(target_channel, VoiceChannel):
+                if voice_chat is None or voice_chat == source_channel:
+                    return
+
+                await voice_chat.send(
+                    message.replace("{duration}", str(duration)).replace("current", target_channel.mention)
+                )
+
+            else:
+                await target_channel.send(message.replace("{duration}", str(duration)))
+
     @commands.command(aliases=("hush",))
     @lock_arg(LOCK_NAMESPACE, "ctx", attrgetter("channel"), raise_error=True)
```

Comment:
I don't follow what you mean here. The function is only responsible for sending alerts to the source and target channel (replacing the target channel with a voice chat channel in case of VC). Choosing between the invocation channel and a channel argument was left up to the caller.

There is one check that is sort of related to that, the one that compares source to target channel to prevent duplicate messages.

Reply:
The `lock_arg` is locking on `ctx.channel`. This is no longer adequate since you've added a channel arg that may be specified instead of using the context. Therefore, the lock decorator needs to be adjusted so it can lock on the correct channel.